Datadog Agent configurable timeout while collecting logs with the docker daemon (#7159)
DataDog/documentation: content/en/logs/guide/docker-logs-collection-troubleshooting-guide.md

4. Mount the YAML file into your container following the instructions in the [Docker Agent documentation][3]. For more information on setting log drivers for Docker containers, [see this documentation][4].

> ## Logs are missing but only for some containers

Reviewer: I'm not sure this solution is actually the most common one for the issue of `Logs are missing but only for some containers` (in most cases, I'd expect this issue to be related to misconfiguration).

Let's rename this to:
```suggestion
## Agent doesn't send logs from containers that have persisted a large volume of logs (> 1GB)
```

What do you think?

> The Docker daemon can have performances issues while it is trying to retrieve logs for containers for which it has already stored large logs files on disk. This could lead to read timeouts when the Logs Agent is gathering the containers' logs from the Docker daemon. The Logs Agent outputs a log containing `Restarting reader after a read timeout` when it occurs.

Reviewer 2: ```suggestion
The Docker daemon can have performances issues while it is trying to retrieve logs for containers for which it has already stored large logs files on disk. This could lead to read timeouts when the Datadog Agent is gathering the containers' logs from the Docker daemon. 
The Datadog Agent outputs a log containing `Restarting reader after a read timeout` when it occurs.
```

> The default read timeout is set to 30 seconds, increasing this value gives more time to the Docker daemon to reply to the Logs Agent. This value can be set in `datadog.yaml` using the `docker_client_read_timeout` parameter or by using the environment variable `DD_LOGS_CONFIG_DOCKER_CLIENT_READ_TIMEOUT`, please note that this value is a duration in seconds. Here is an example increasing it to 60 seconds:

Reviewer 2: ```suggestion
The default read timeout is set to 30 seconds, increasing this value gives more time to the Docker daemon to reply to the Datadog Agent. This value can be set in `datadog.yaml` using the `logs.config.docker_client_read_timeout` parameter or by using the environment variable `DD_LOGS_CONFIG_DOCKER_CLIENT_READ_TIMEOUT`, please note that this value is a duration in seconds. Here is an example increasing it to 60 seconds:
```

4. Mount the YAML file into your container following the instructions in the [Docker Agent documentation][3]. For more information on setting log drivers for Docker containers, [see this documentation][4].

## Logs are missing but only for some containers

The Docker daemon can have performances issues while it is trying to retrieve logs for containers for which it has already stored large logs files on disk. This could lead to read timeouts when the Datadog Agent is gathering the containers' logs from the Docker daemon. 

> The Datadog Agent outputs a log containing `Restarting reader after a read timeout` when it occurs.

Reviewer: there are other reasons the Logs agent may log this, generally related to some limitations of the Docker daemon that the Logs Agent tries to work around. Examples I can think of:

* Docker daemon rotating the log file, when the docker log driver's max-file setting is set to 1, leads to a timeout as well. In this example, increasing the timeout on the Logs Agent is actually counter-productive and makes the problem worse.
* container is actually not logging anything for 30+ seconds. I _think_ this leads to a timeout as well (but transparent to the user)

Therefore let's give a more accurate description of the issue: here we could say that the symptoms are that the Logs Agent outputs this message every 30 seconds for a given container and stops sending logs from that container while the container is actually logging messages.

The default read timeout is set to 30 seconds, increasing this value gives more time to the Docker daemon to reply to the Datadog Agent. This value can be set in `datadog.yaml` using the `logs_config.docker_client_read_timeout` parameter or by using the environment variable `DD_LOGS_CONFIG_DOCKER_CLIENT_READ_TIMEOUT`, please note that this value is a duration in seconds. Here is an example increasing it to 60 seconds:

> ```text

Reviewer: ```suggestion
```yaml
```

(assuming it's supported on the docs website?)

4. Mount the YAML file into your container following the instructions in the [Docker Agent documentation][3]. For more information on setting log drivers for Docker containers, [see this documentation][4].

## Agent doesn't send logs from containers that have persisted a large volume of logs (> 1GB)

The Docker daemon can have performances issues while it is trying to retrieve logs for containers for which it has already stored large logs files on disk. This could lead to read timeouts when the Datadog Agent is gathering the containers' logs from the Docker daemon. 

When it occurs, the Datadog Agent outputs a log containing `Restarting reader after a read timeout` for a given container every 30 seconds and stops sending logs from that container while it is actually logging messages.

> The default read timeout is set to 30 seconds, increasing this value gives more time to the Docker daemon to reply to the Datadog Agent. This value can be set in `datadog.yaml` using the `logs_config.docker_client_read_timeout` parameter or by using the environment variable `DD_LOGS_CONFIG_DOCKER_CLIENT_READ_TIMEOUT`, please note that this value is a duration in seconds. Here is an example increasing it to 60 seconds:

Reviewer 3: ```suggestion
The default read timeout is set to 30 seconds, increasing this value gives more time to the Docker daemon to reply to the Datadog Agent. This value can be set in `datadog.yaml` using the `logs_config.docker_client_read_timeout` parameter or by using the environment variable `DD_LOGS_CONFIG_DOCKER_CLIENT_READ_TIMEOUT`. This value is a duration in seconds, find below an example increasing it to 60 seconds:
```
